nixos/modules/services/networking/clash-meta.nix

```
@@ -0,0 +1,129 @@
+# NOTE:
+# this module also applies to `clash` and *maybe* other clash derivives,
+# but `clash`'s repo was deleted and removed from nixpkgs a while ago, so I did s/clash/clash-meta/g
+#
+# NOTE:
+# `clash-meta` is now renamed to `mihomo`, please update the name after this PR is merged:
+# https://github.com/NixOS/nixpkgs/pull/271658/
+#
+# NOTE:
+# cfg.configFile contains secrets such as proxy servers' credential!
+# we dont want plaintext secrets in world-readable `/nix/store`.
+#
+# NOTE:
+# cfg.tunMode is `clash-premium` (not in nixpkgs, repo removed, unfree) and `clash-meta` specific feature.
+
+{ lib
+, config
+, pkgs
+, ...
+}:
+let
+  cfg = config.services.clash-meta;
+in
+{
+  options.services.clash-meta = {
+    enable = lib.mkEnableOption "Clash.Meta, A rule-based proxy in Go.";
+
+    package = lib.mkPackageOption pkgs "clash-meta" { };
+
+    configFile = lib.mkOption {
+      default = null;
+      type = lib.types.nullOr lib.types.path;
+      description = "Configuration file to use.";
+    };
+
+    webui = lib.mkOption {
+      default = null;
+      type = lib.types.nullOr lib.types.path;
+      description = ''
+        Local web interface to use.
+
+        You can also use the following website, just in case:
+        - metacubexd:
+          - https://d.metacubex.one
+          - https://metacubex.github.io/metacubexd
+          - https://metacubexd.pages.dev
+        - yacd:
+          - https://yacd.haishan.me
+        - clash-dashboard (buggy):
+          - https://clash.razord.top
+      '';
+    };
+
+    extraOpts = lib.mkOption {
+      default = null;
+      type = lib.types.nullOr lib.types.str;
+      description = "Extra command line options to use.";
+    };
+
+    tunMode = lib.mkEnableOption ''
+      necessary permission for Clash.Meta's systemd service for TUN mode to function properly.
+
+      Keep in mind, that you still need to enable TUN mode manually in Clash.Meta's configuration.
+    '';
+  };
+
+  config = lib.mkIf cfg.enable {
+    ### systemd service
+    systemd.services."clash-meta" = {
+      description = "Clash.Meta daemon, A rule-based proxy in Go.";
+      documentation = [ "https://wiki.metacubex.one/" ];
+      requires = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig =
+        {
+          ExecStart = builtins.concatStringsSep " " [
```

```suggestion
          ExecStart = lib.concatStringsSep " " [
```

For better compatibility, I guess?